contracts/MyToken_721.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract MyToken_721 is ERC721Enumerable, Ownable, ReentrancyGuard {
+
+    uint private _tokenPrice;
+    string private _tokenBaseUri;
+    mapping(address => uint) private _minters; 
+
+    constructor(string memory baseUri, uint tokenPrice) ERC721("MyToken", "MTK") {
+        _tokenBaseUri = baseUri;
+        _tokenPrice = tokenPrice;
+    }
+
+    function safeMint() public payable nonReentrant {
+        uint mintCount = _minters[msg.sender];
+        if (mintCount > 9) {
+            require(msg.value >= _tokenPrice, "Token price is less!");
+        } else {
+            require(msg.value == 0, "For 10 Tokens there is no cost!");
+        }
+        _minters[msg.sender] += 1;
+        _safeMint(msg.sender, totalSupply() + 1);
+    }
+
+    function _baseURI() internal view override virtual returns (string memory) {
+        return _tokenBaseUri;
+    }
+
+    function changeBaseURI(string calldata baseUri) external onlyOwner  {
+        _tokenBaseUri = baseUri;
+    }
+
+    function balance() external view returns (uint) {

```

Comment:
I think no need to have this function as it is possible to get contract balance by web3 functions.